pubsublite/spark-connector/noxfile_config.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Default TEST_CONFIG_OVERRIDE for python repos.
+
+# You can copy this file into your directory, then it will be imported from
+# the noxfile.py.
+
+# The source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/noxfile_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    # NOTE: Apache Beam does not currently support Python 3.9.
+    "ignored_versions": ["2.7", "3.6", "3.7", "3.9"],
```

Comment:
Think I missed this before, but is it a strict requirement to only be using 3.8?

Reply:
I can open this up as a last step. Right now just want to test to run fast.

Reply:
@jiangmichaellll Looks like multiple streaming jobs in the same cluster will produce "Multiple streaming jobs detected for 734": https://pantheon.corp.google.com/dataproc/jobs/0f418c4d-e2ae-49b9-ad09-598f8a2f559c/monitoring?region=us-west1&project=python-docs-samples-tests

Since we are just using one Spark cluster, I vote we just test against one Python version.

Reply:
hmm did you manually run jobs concurrently? I never seen this before. I usually either change app name or only run one at a time.

Reply:
When two Python versions are selected, two ITs run in parallel (two different topics created at roughly the same time, two PySpark jobs submitted at the roughly the same time as well).

Reply:
Used uuid in appName when starting up a SparkSession.

Reply:
Using uuid in `checkpointLocation` solved `"Concurrent update to the log`.

Reply:
But occasionally you may see something like [this](https://pantheon.corp.google.com/dataproc/jobs/19cfb30c-bbe8-3f7c-b4e5-5be6fab79e50/monitoring?region=us-west1&project=python-docs-samples-tests):

```
21/09/16 19:19:53 INFO com.google.cloud.pubsublite.spark.PslSparkUtils: Input schema to write to Pub/Sub Lite doesn't contain attributes column, this field for all rows will be set to empty. [CONTEXT ratelimit_period="5 MINUTES" ]
21/09/16 19:19:54 INFO com.google.cloud.pubsublite.spark.PslStreamWriter: Committed 0 messages for epochId:0.
21/09/16 19:19:54 WARN org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor: Current batch is falling behind. The trigger interval is 1000 milliseconds, but spent 1334 milliseconds
21/09/16 19:20:09 WARN org.apache.spark.scheduler.cluster.YarnScheduler: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
21/09/16 19:20:24 WARN org.apache.spark.scheduler.cluster.YarnScheduler: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
21/09/16 19:20:39 WARN org.apache.spark.scheduler.cluster.YarnScheduler: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
21/09/16 19:20:52 ERROR org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec: Data source writer org.apache.spark.sql.execution.streaming.sources.MicroBatchWriter@6d4c678c is aborting.
21/09/16 19:20:52 WARN com.google.cloud.pubsublite.spark.PslStreamWriter: Epoch id: 1 is aborted, 0 messages might have been published.
```